parser/src/parser.rs

```diff
@@ -636,6 +636,79 @@ class Foo(A, B):
         insta::assert_debug_snapshot!(ast::Suite::parse(source, "<test>").unwrap());
     }
 
+
+    #[test]
+    #[cfg(feature = "all-nodes-with-ranges")]
+    fn test_parse_class_with_empty_generic() {
+        let source = "\
+class Foo[](A, B):
+  pass
+";
+        insta::assert_debug_snapshot!(ast::Suite::parse(source, "<test>").unwrap());
+    }
```

Comment:
Failing test currently; need to determine if this is valid syntax in Python 3.12

Reply:
Invalid syntax

```
>>> class Foo[]:
  File "<stdin>", line 1
    class Foo[]:
              ^
```

Reply:
Removed in https://github.com/RustPython/Parser/pull/95/commits/7952380f720463108c0798ea5649e974251d2d38